pywhat/Data/regex.json

```diff
@@ -212,7 +212,7 @@
     },
     {
       "Name": "JSON Web Token (JWT)",
-      "Regex": "^([a-zA-Z0-9_=]+)\\.([a-zA-Z0-9_=]+)\\.([a-zA-Z0-9_\\-\\+\/=]*)",
+      "Regex": "(?i)^(?=.*[a-z])(?=.*[0-9])(?:[a-z0-9_=]+\\.){2}(?:[a-z0-9_\\-\\+\/=]*)$",

```

Comment:
If this was YAML we could totally write a comment here explaining this.... >:(

Reply:
I can break it down here! :smile:  So `(?i)` means case-insensitive (that is why I removed A-Z). `(?=.*[a-z])` means look-ahead - basically it is checking if in the whole match there is at least one letter from a-z. Then there is this `(?:[a-z0-9_=]+\\.)` which is the first part of the JWT. As first and second parts are the same, I added them together so there is now `(?:[a-z0-9_=]+\\.){2}` - two times. And then the last part. I also added `$` at the end as it was missing, so boundary was not working here - that is the main reason why it matched to a lot more than just JWT. If first few characters seemed like a JWT, it would match :D

Reply:
Ohhh that is unfortunate that you cannot add comments yeah :/